notebooks/403-action-recognition-webcam/403-action-recognition-webcam.ipynb

```diff
@@ -0,0 +1,691 @@
+    "    :param top_k: The k most probable positions in the list of labels\n",
+    "    :returns: decoded_labels: The k most probable actions from the labels list\n",
+    "              decoded_top_probs: confidence for the k most probable actions   \n",
+    "    \"\"\"\n",
+    "    top_ind = np.argsort(-1 * probs)[:top_k]\n",
+    "    out_label = np.array(labels)[top_ind.astype(int)]\n",
+    "    decoded_labels = [out_label[0][0], out_label[0][1], out_label[0][2]]\n",
+    "    top_probs = np.array(probs)[0][top_ind.astype(int)]\n",
+    "    decoded_top_probs = [top_probs[0][0], top_probs[0][1], top_probs[0][2]]\n",
+    "    return decoded_labels, decoded_top_probs\n",
+    "\n",
+    "\n",
+    "# Configuration for displaying images with text\n",
+    "FONT_COLOR = (255, 255, 255)\n",
+    "FONT_STYLE = cv2.FONT_HERSHEY_DUPLEX\n",
+    "FONT_SIZE = 0.7\n",
+    "TEXT_VERTICAL_INTERVAL = 25\n",
+    "TEXT_LEFT_MARGIN = 15           "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### AI funtions\n",
+    "\n",
+    "Following this pipeline, \n",
+    "\n",
+    "<img align='center' src=\"action_recognition_pipeline.jpeg\" alt=\"drawing\" width=\"1000\"/>\n",
+    "\n",
+    "\n",
+    "we will use the next functions to help us to:\n",
+    "\n",
+    "1. Preprocess frame before running the Encoder. (`preprocessing`)\n",
+    "2. Encoder Inference per frame. (`encoder`)\n",
+    "3. Decoder inference per set of frames. (`decoder`)\n",
+    "4. Normalize the Decoder output to get confidence values per action recognition label. (`softmax`)\n",
+    "\n",
+    "Also, we can see here the minimal configuration to display text over the live streaming or video for the main process."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "def preprocessing(frame: np.ndarray, size: int) -> np.ndarray:\n",
+    "    \"\"\"\n",
+    "    Preparing frame before Encoder.\n",
+    "    The image should be scaled to its shortest dimension at \"size\" \n",
+    "    and cropped, centered, and squared so that both width and \n",
+    "    height have lengths \"size\". Frame must be transposed from \n",
+    "    Height-Width-Channels (HWC) to Channels-Height-Width (CHW).\n",
+    "    :param frame: input frame \n",
+    "    :param size: input size to encoder model\n",
+    "    :returns: resized and cropped frame\n",
+    "    \"\"\"\n",
+    "    # Adaptative resize\n",
+    "    preprocessed = adaptive_resize(frame, size)\n",
+    "    # Center_crop                \n",
+    "    preprocessed = center_crop(preprocessed)\n",
+    "    # Transpose frame HWC -> CHW\n",
+    "    preprocessed = preprocessed.transpose((2, 0, 1))  # HWC -> CHW\n",
+    "    return preprocessed\n",
+    "\n",
+    "\n",
+    "def encoder(preprocessed: np.ndarray, exec_net_en: ExecutableNetwork, input_key_en: str, encoder_output: list) -> list:\n",
```

Comment:
Do you mean I don't need to use arguments for encoder and decoder functions?

Reply:
I don't want to remove arguments. I prefer to see a clear function with input arguments and returns. It is clear to see what kind of information is necessary. It could help to understand better the pipeline.